### ISSUE TITLE
Provide option to customize automatic/background update interval

### ISSUE THREAD

**Issue by farmerlandrose** [accrescent/accrescent#575]
Accrescent currently seems to do automatic app update checks multiple times a day. Please add a configuration setting to adjust this automatic update interval akin to other stores/downloaders like Obtainium or F-Droid client. 

Example values: Every x hours, Daily, Every x days

Thank you and kind regards

Loosely Related: https://github.com/accrescent/accrescent/issues/545

**Comment by PatrykMis** [comment]
I agree, as currently we don't even know how many times a day the update check is being performed (without checking the source code, of course).

**Comment by lberrymage** [comment]
I realize this is often a feature in updater-type apps, but is there a specific reason to allow changing the update check interval? I don't see the benefit.

**Comment by farmerlandrose** [comment]
Hi @lberrymage,
yes, there are privacy reasons. This does not mean I distrust Accrescent. Though I like to have more control over my outgoing connections in general, especially when they are done periodically. Raising the update check interval to days/week in my view is a good trade-off between convenience and the amount of background connections initiated. In addition this alternative seems to be safer than disabling auto-checks all together, which contains the risk to forget manual app updates.

**Comment by farmerlandrose** [comment]
I still would be curious to know, if custom auto-update intervals will be a planned/supported feature in future. What is your stance on this @lberrymage ?

**Comment by lberrymage** [comment]
I'm hesitant because of the lack of justification for the feature. It wouldn't be difficult to add, but I could see it making accurate metrics collection more difficult from the operations side. It's also not clear to me how changing the update interval has any definitive privacy (or other) benefits; it could even have privacy _downsides_ since users which use a non-default interval would be the minority and so could be separately identified.

In short, it has no clear benefits from my perspective and many potential downsides, so I'm currently disinclined to support it.

**Comment by farmerlandrose** [comment]
Thanks for describing your concerns, I am commenting below.

>  It's also not clear to me how changing the update interval has any definitive privacy (or other) benefits

The general goal is data minimization and control of connections. Configuring auto-update interval or manual update checks is part of it. 
While Accrescent certainly is just one piece of the puzzle, if you have say 100 apps, and each of them polls hourly for updates or does other things in the background, connections simply add up to the point of not being manageable anymore.
Other app stores/downloaders like GrapheneOS App Store, Obtainium, F-Droid all provide customization options for update checking. I'd imagine, implementation might be beneficial for broader acceptance of Accrescent as well.

> it could even have privacy downsides since users which use a non-default interval would be the minority and so could be separately identified.

Yeah, certainly hard to answer. Please at least give users a choice to decide on their own.

To reiterate suggestions:
- Provide a toggle "Check for updates" or similar. If disabled, it is the user's responsibility to manually check for updates. Currently Accrescent also seems to check for updates, if "Automatic updates" is disabled, so I assume this option is just about installing, not checking.
- Provide an "Update check interval" with 1/2/6/12/24 hours, 3 days/ weekly or similar. This setting is only used, if "Check for updates" is enabled.

> It wouldn't be difficult to add, but I could see it making accurate metrics collection more difficult from the operations side.

If metrics collection is the only downside: What metrics are you collecting and how is it related to users' update interval decisions?

**Comment by Frmd-git** [comment]
In my opinion, following the example of the GrapheneOS store with them having customizable intervals as well makes a lot of sense.
